tests/test_doc_saver.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import os
+
+import pytest
+import redis
+from langchain_core.documents.base import Document
+
+from langchain_google_memorystore_redis.doc_saver import MemorystoreDocumentSaver
+
+
+@pytest.mark.parametrize(
+    "page_content,metadata,content_field,metadata_fields",
+    [
+        (
+            '"content1"',
+            {"key1": "doc1_value1", "key2": "doc1_value2"},
+            "page_content",
+            None,
+        ),
+        (
+            '"content2"',
+            {"key1": {'"nested_key"': {'"double_nested"': '"doc2_value1"'}}},
+            "special_page_content",
+            None,
+        ),
+        (
+            '"content3"',
+            {"key1": {"k": "not_in_filter"}, "key2": {'"key"': "in_filter"}},
+            "page_content",
+            set(["key2"]),
+        ),
+    ],
+)
+def test_doc_saver_add_documents_one_doc(
```

Comment:
curious - who calls this function?

Reply:
The integration test [does](https://github.com/googleapis/langchain-google-memorystore-redis-python/blob/staging/integration.cloudbuild.yaml#L21-L26). You can also invoke it by `REDIS_URL=redis://localhost:6379 python -m pytest` if you have a redis server listening on localhost:6379